pkgs/misc/vscode-extensions/vscode-lldb/default.nix

```
@@ -1,106 +1,101 @@
 
-  vsix = stdenv.mkDerivation {
-    name = "${name}-${version}-vsix";
-    inherit src;
+in stdenv.mkDerivation rec {
+  name = "vscode-extension-${pname}";
+  inherit src;
+  vscodeExtUniqueId = "${publisher}.${pname}";
+  installPrefix = "share/vscode/extensions/${vscodeExtUniqueId}";
 
-    # Only build the extension. We handle `adapter` and `lldb` with nix.
-    patches = [ ./cmake-build-extension-only.patch ];
+  nativeBuildInputs = [ cmake nodejs unzip makeWrapper ];
 
-    nativeBuildInputs = [ cmake nodejs unzip breakpointHook ];
+  patches = [ ./cmake-build-extension-only.patch ];
 
-    postConfigure = ''
-      cp -r ${build-deps}/lib/node_modules/vscode-lldb/{node_modules,package-lock.json} .
-    '';
+  postConfigure = ''
+    cp -r ${nodeDeps}/lib/node_modules/vscode-lldb/{node_modules,package-lock.json} .
+  '';
 
-    makeFlags = [ "vsix_bootstrap" ];
+  cmakeFlags = [
+    # Do not append timestamp to version.
+    "-DVERSION_SUFFIX="
+  ];
+  makeFlags = [ "vsix_bootstrap" ];
 
-    installPhase = ''
-      unzip ./codelldb-bootstrap.vsix 'extension/*' -d ./vsix-extracted
-      mv vsix-extracted/extension $out
+  installPhase = ''
+    ext=$out/$installPrefix
+    runHook preInstall
 
-      ln -s ${adapter}/adapter $out
-      # Mark that adapter and lldb are installed.
-      touch $out/platform.ok
-    '';
+    unzip ./codelldb-bootstrap.vsix 'extension/*' -d ./vsix-extracted
 
-    dontStrip = true;
-    dontPatchELF = true;
-  };
+    mkdir -p $ext/{adapter,formatters}
+    mv -t $ext vsix-extracted/extension/*
+    cp -t $ext/adapter ${adapter}/{bin,lib}/* ../adapter/*.py
+    cp -t $ext/formatters ../formatters/*.py
+    ln -s ${lldb} $ext/lldb
+    # Mark that all components are installed.
+    touch $ext/platform.ok
 
-in vscode-utils.buildVscodeExtension {
-  inherit name;
-  src = vsix;
+    runHook postInstall
+  '';
 
-  nativeBuildInputs = lib.optional setDefaultLldbPath jq;
-  postUnpack = lib.optionalString setDefaultLldbPath ''
-    jq '.contributes.configuration.properties."lldb.library".default = $s' \
-      --arg s "${lldb}/lib/liblldb.so" \
-      $sourceRoot/package.json >$sourceRoot/package.json.new
-    mv $sourceRoot/package.json.new $sourceRoot/package.json
+  # `adapter` will find python binary and libraries at runtime.
+  fixupPhase = ''
+    wrapProgram $out/$installPrefix/adapter/codelldb \
+      --prefix PATH : "${python3}/bin" \
+      --prefix LD_LIBRARY_PATH : "${python3}/lib"
   '';
 
-  vscodeExtUniqueId = "${publisher}.${name}";
+  passthru = {
+    inherit lldb adapter;
+  };
 
   meta = with lib; {
     description = "A native debugger extension for VSCode based on LLDB";
     homepage = "https://github.com/vadimcn/vscode-lldb";
     license = with licenses; [ mit ];
     maintainers = with maintainers; [ oxalica ];
-    platforms = platforms.all;
+    # Build failed on x86_64-darwin.
+    platforms = [ "x86_64-linux" ];
```

Build failures should be represented by broken and platforms should be what upstream theoretically supports. There is a darwin release upstream so this should be broken https://github.com/vadimcn/vscode-lldb/releases.

```suggestion
    platforms = platforms.all;
    broken = stdenv.isDarwin;
```

---

@SuperSandro2000 Got it. Applied.